crates/services/shared-sequencer/src/service.rs

```diff
@@ -0,0 +1,262 @@
+where
+    S: Signer + 'static,
+{
+    const NAME: &'static str = "SharedSequencer";
+
+    type SharedData = EmptyShared;
+    type Task = Task<S>;
+    type TaskParams = ();
+
+    fn shared_data(&self) -> Self::SharedData {
+        EmptyShared
+    }
+
+    async fn into_task(
+        mut self,
+        _: &StateWatcher,
+        _: Self::TaskParams,
+    ) -> anyhow::Result<Self::Task> {
+        let shared_sequencer_client = if let Some(endpoints) = &self.config.endpoints {
+            let ss = Client::new(endpoints.clone(), self.config.topic).await?;
+
+            if self.signer.is_available() {
+                let cosmos_public_address = ss.sender_account_id(self.signer.as_ref())?;
+
+                tracing::info!(
+                    "Shared sequencer uses account ID: {}",
+                    cosmos_public_address
+                );
+            }
+
+            Some(ss)
+        } else {
+            None
+        };
+
+        let blobs = Arc::new(tokio::sync::Mutex::new(SSBlobs::new()));
+
+        if self.config.enabled {
+            let mut block_events = self.blocks_events;
+
+            tokio::task::spawn({
+                let blobs = blobs.clone();
+                async move {
+                    while let Some(block) = block_events.next().await {
+                        let blob = SSBlob {
+                            block_height: *block.sealed_block.entity.header().height(),
+                            block_id: block.sealed_block.entity.id(),
+                        };
+                        blobs.lock().await.push(blob);
+                    }
+                }
+            });
+        }
+
+        Ok(Task {
+            shared_sequencer_client,
+            config: self.config,
+            signer: self.signer,
+            account_metadata: None,
+            prev_order: None,
+            blobs,
+        })
+    }
+}
+
+impl<S> Task<S>
+where
+    S: Signer,
+{
+    async fn blobs(&mut self) -> anyhow::Result<Option<SSBlobs>> {
+        let ss = self
+            .shared_sequencer_client
+            .as_ref()
+            .expect("Shared sequencer client is not set; qed");
+
+        if self.account_metadata.is_none() {
+            // If the account is not funded, this code will fail
+            // because we can't sign the transaction without account metadata.
+            let account_metadata = ss.get_account_meta(self.signer.as_ref()).await;
+
+            match account_metadata {
+                Ok(account_metadata) => {
+                    self.account_metadata = Some(account_metadata);
+                }
+                Err(err) => {
+                    tokio::time::sleep(Duration::from_secs(1)).await;
+                    return Err(err);
+                }
+            }
+        }
+
+        if self.prev_order.is_none() {
+            self.prev_order = ss.get_topic().await?.map(|f| f.order);
+        }
+
+        tokio::time::sleep(self.config.block_posting_frequency).await;
```

Comment:
We might prefer a interval-based in the `select` solution because if this is cancel and another branch resolves this sleep will reset and we can endup to not publish blocks for hours. This can not happen now because the select doesn't have any other logic (except the service one) and so can't be cancelled. But this coupling in the future might be problematic. (except if we are ok with the sleep reset)